### ISSUE TITLE
Error in dispersion

### ISSUE THREAD

**Issue by retide** [PolMine/polmineR#23]
Dear Andreas, the dispersion function does not work for me:

I checked whether the corpus and sAttributes are OK:

```
> sAttributes("AFD")
[1] "s"           "p"           "text"        "text_id"     "text_source" "text_file"   "text_url"    "text_date"   "text_year"  

> count("AFD", "Krise")
   query count         freq
1: Krise   105 0.0001310295

> size("AFD", sAttribute = "text_year")
... cwb-s-decode utility available, going to use it
   text_year   size
1:      2009  59741
2:      2010 134472
3:      2011  91151
4:      2012 111813
5:      2013 122815
6:      2014  60762
7:      2015  57929
8:      2016 104417
9:      2017  58246
```

But dispersion throws an error:

```
> dispersion("AFD", "Krise", sAttribute = "text_year")
  |======================================================================================================================================| 100%
Fehler in f(x[[i]], mc = FALSE, progress = FALSE, verbose = FALSE, ...) : 
  unbenutzte Argumente (mc = FALSE, progress = FALSE, verbose = FALSE)
```

Best,
Thilo

**Comment by ablaette** [comment]
Dear Thilo,
I do not manage to reproduce the error with the corpora I have at hand. The dispersion method has seen some refactoring, and yes, it did not always work. Can you check that you have the most recent version of polmineR from the development/dev branch. Please let me know whether the error still turns up.
```r
devtools::install_github("PolMine/polmineR", ref = "dev")
```

**Comment by retide** [comment]
Dear Andreas,

I am on the most recent version.
```
> devtools::install_github("PolMine/polmineR", ref = "dev")
Skipping install of 'polmineR' from a github remote, the SHA1 (1167ac34) has not changed since last install.
  Use `force = TRUE` to force installation
```

The same error occurs for another corpus on my Macbook. On our Ubuntu server, however, dispersion works fine. I am not 100 percent sure that there is no error in the corpora I am using on my Macbook. I will rule that out first and let you know.

**Comment by ablaette** [comment]
It would be a strange error I have not yet come across. I tried to reproduce the error on a Macbook, so it should not be a Mac/Ubuntu thing.

One potential source of errors with the dispersion method has been missing values for certain years. To check that, you might try to create partitions for the individual years in the corpus and do a count of 'Krise' for these partitions.

Let me know the result of your checks!

**Comment by retide** [comment]
I tried with a corpus identical to the corpus on the server, but the error still occurs on my macbook:

```
> dispersion("TAGESSCHAU", "Krise", sAttribute = "text_year")
  |=====================================================================================================| 100%
Fehler in f(x[[i]], mc = FALSE, progress = FALSE, verbose = FALSE, ...) : 
  unbenutzte Argumente (mc = FALSE, progress = FALSE, verbose = FALSE)
```

Creating partitions for all attribute values and doing counts on those works fine:

```
> s <- size("TAGESSCHAU", sAttribute = "text_year")
... cwb-s-decode utility available, going to use it
> for(y in s$text_year) {
+   p <- partition("TAGESSCHAU", text_year=y)
+   print(count(p, "Krise"))
+ }
Setting up partition 
... get encoding: UTF-8
... get cpos and strucs
... get partition size
... processing query Krise
   query count         freq
1: Krise   878 0.0001998647
Setting up partition 
... get encoding: UTF-8
... get cpos and strucs
... get partition size
... processing query Krise
   query count         freq
1: Krise   742 0.0001279568
Setting up partition 
... get encoding: UTF-8
... get cpos and strucs
... get partition size
... processing query Krise
   query count        freq
1: Krise   321 0.000125299
Setting up partition 
... get encoding: UTF-8
... get cpos and strucs
... get partition size
... processing query Krise
   query count freq
1: Krise     0    0
```

**Comment by retide** [comment]
I got it working. For rather historical reasons I was running R within an anaconda environment. This hasn't caused any trouble thus far, but since I shifted most of my text analysis experiments from python to R it is also no longer necessary. I reinstalled polmineR and all dependencies in my default user environment and there dispersion works fine. 

I am not sure whether it is worth digging into the error as I guess it may have been the result of some awkward function masking or dependency version conflict that few (if any) other people will actually encounter. But let me know in case you would like me to do some further testing.
